src/frontend/src/utils/userIntent.ts

```diff
@@ -8,7 +8,7 @@ export type AddDeviceIntent = { kind: "addDevice" };
 export const intentFromUrl = (url: URL): UserIntent => {
   if (url.hash == "#authorize") {
     return { kind: "auth" };
-  } else if (url.hash?.split("device=")[1]) {
+  } else if (url.hash !== ""?.split("device=")[1]) {
```

Comment:
This looks weird. The `?` here is not the ternary operator, but a part of the null-safe access operator `?.`.

Reply:
Oh damn you're right!

Reply:
PTAL!